packages/main/src/Menu.ts

```diff
@@ -338,8 +390,22 @@ class Menu extends UI5Element {
 		}, MENU_OPEN_DELAY);
 	}
 
+	_isGroupParent(parent: MenuItem | Menu | MenuItemGroup): boolean {
+		return parent && "isGroup" in parent && parent.isGroup;
+	}
+
 	_itemClick(e: CustomEvent<ListItemClickEventDetail>) {
 		const item = e.detail.item as MenuItem;
+		const prevSelected = item.isSelected;
+		const parent = item.parentElement as MenuItem | Menu | MenuItemGroup;
+
+		if (this._isGroupParent(parent)) {
```

Comment:
I would suggest we remove the "isGroupParent" method, as well as this logic here. And remove any knowledge about the parent element. Create an event that will notify the group that Item was selected and execute this logic in the group. Because currently we look for an element, check if it's a group then work with groups methods and properties, if we do it with event, only the components that need to know that a selection occured will listen for this event and will execute functions based on the information.

Reply:
Done